src/util/QCompareCaseInsensitive.hpp

```diff
@@ -0,0 +1,144 @@
+#pragma once
+
+#include <QLatin1String>
+#include <QString>
+#include <QStringView>
+#include <QtGlobal>
+
+#if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
+#    include <QUtf8StringView>
+#endif
+
+namespace chatterino {
+
+/// Case insensitive transparent comparator for Qt's string types
+struct QCompareCaseInsensitive {
```

Comment:
I'm going to use this for [`LinkParser`](https://github.com/Chatterino/chatterino2/blob/fdecb4a39f3e0340fe555797030454986b50be78/src/common/LinkParser.cpp#L46) in a future PR to remove the last allocation it does while parsing.